### PR TITLE
Update DG and include copy description

### LINKED ISSUE #245: Update and fix DG bugs

### PR DESCRIPTION
**Pull request by jeremiahlzz**
* fix minor format mistakes
* fix multiplicity of `day` in `timetable`
* add `copy` command implementation
* add copy sequence diagram
* add edit sequence diagram
* add details for manual testing
* add future feature plans
* update version number to v1.4
* add URL regex section under Website URL support

Resolves #245
Future features added to DG #201, #208, #224, #230, #231, #240, #257

### CODE REVIEW

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 1]
```diff
@@ -20,6 +20,6 @@ Student *--> "1" TelegramHandle
 Student *--> "1" Email
 Student *--> "0..1" Link
 Student *--> "1" Timetable
-Timetable *--> "1..7" Day
+Timetable *--> "5..7" Day
```
Comment (darryl-chan) [inline docs/diagrams/BetterModelClassDiagram.puml:23]: Should this be 5 or 7?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -624,7 +702,21 @@ finding common slots and copying student emails, which showcases our goal of cre
...
+
+7. **Limits to Telegram Handle, Email and Tags:** There is currently no character limit for the `Telegram Handle`, `Email` and `Tag` fields. We plan to limit these fields to an appropriate character count to better represent these fields. Example:
+   1. Telegram Handle maximum length is 32 characters according to [here](https://limits.tginfo.me/en).
+   2. Email maximum length is 320 characters.
+   3. Tag should be sufficiently long but not too long as they are used as groupings. We estimate around 40 characters to be more than sufficient.
+
+8. **Editing Student with the same details:** Currently, when editing a student with the same details, no error message is displayed and EduConnect updates the Student with the same data. This is may confuse users and it will result in no changes while still showing a successfuull update. We plan to add a check and display the proper error message. E.g. A student with telegram handle `@bunny`, when running the command `edit h:@bunny h/@bunny` should return an error message `Duplicate telegram handle supplied.`
```
Comment (fy17ohhh) [inline docs/DeveloperGuide.md:722]: Remember also to include the future enhancement for sat/sun timetable:)

**Comment by jeremiahlzz** [comment]
## Edit Sequence Diagram
image

## CopySequenceDiagram
image

**Comment by darryl-chan** [comment]
LGTM

**Pull request merged**